Add support for IngressClass [#5232]

andreadecorte opened the pull request:
Adding support for Ingress Classes. I added them under "Cluster" section, let me know if you see them elsewhere instead (under Service?)

![image](https://user-images.githubusercontent.com/104198/159805606-21bf3ed8-c27e-4c24-ae52-1997219428af.png)
![image](https://user-images.githubusercontent.com/104198/159805633-b21b31fe-1df2-4f03-aac6-c1d1e89d7fc1.png)

floreks commented:
Our layout is based on the https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#ingressclass-v1-networking-k8s-io

IngressClass is under Service section which means it should be the same in the Dashboard.

andreadecorte commented:
Thanks @floreks , moved to the Service section, updated screenshot below
![Screenshot 2022-03-24 at 11 17 08](https://user-images.githubusercontent.com/104198/159897026-9c057a54-08f4-4962-9c9c-8ba776b002ff.png)

floreks commented on src/app/frontend/typings/root.api.ts:
```diff
@@ -405,6 +409,11 @@ export interface StorageClass extends Resource {
 
+export interface IngressClass extends Resource {
+  controller: string;
+  parameters: StringMap[];
```
I don't see this field on the API side.

andreadecorte replied:
I am flattening here https://github.com/kubernetes/dashboard/pull/6920/files#diff-f2a62a475a29db82a76ba9f4ea8915ed8a175af848d689109ecb25e118e6aa22R46 IngressClassParametersReference to a map of strings as all the contents are strings. Should I rather keep a dedicated object for that?

floreks replied:
Ye, but this is not the `IngressClassDetails` object, it's this one https://github.com/kubernetes/dashboard/pull/6920/files#diff-a6cfbdffa7b2f43b890a274cb3a387fcda23797c8cc90e7bada690411a689d19R38-R42

andreadecorte replied:
and you're of course right, field removed from there

floreks commented on src/app/frontend/resource/discovery/ingressclass/routing.ts:
```diff
@@ -0,0 +1,46 @@
+  path: '',
+  component: IngressClassListComponent,
+  data: {
+    breadcrumb: 'Ingress Classes',
```
A new way to localize breadcrumbs should be used here. It was recently merged. There is now `messages.ts` file that localizes such messages.

andreadecorte replied:
thanks, updated and rebased

andreadecorte commented:
did you have any chance to take a look again at this one @floreks ?

floreks reviewed (approved):
/lgtm
/approve

floreks commented:
@andreadecorte I think we can merge. Thanks! :slightly_smiling_face:

andreadecorte commented:
> @andreadecorte I think we can merge. Thanks! 🙂

thank you for the merge!
